restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java

```diff
@@ -998,7 +1000,25 @@ private void onTagMessage(Object message) throws TransitionFailedException, Tran
             } else if (Verbs.record.equals(verb.name())) {
                 fsm.transition(message, creatingRecording);
             } else if (Verbs.sms.equals(verb.name())) {
-                fsm.transition(message, creatingSmsSession);
+
+                //Check if Outbound SMS is allowed
+                ExtensionController ec = ExtensionController.getInstance();
+                final IExtensionFeatureAccessRequest far = new FeatureAccessRequest(FeatureAccessRequest.Feature.OUTBOUND_SMS, accountId);
+                ExtensionResponse er = ec.executePreInboundAction(far, this.extensions);
+
+                if (er.isAllowed()) {
+                    fsm.transition(message, creatingSmsSession);
+                } else {
+                    if (logger.isDebugEnabled()) {
+                        final String errMsg = "Outbound SMS is not Allowed";
+                        logger.debug(errMsg);
+                    }
+                    final NotificationsDao notifications = storage.getNotificationsDao();
+                    final Notification notification = notification(WARNING_NOTIFICATION, 11001, "Outbound SMS is now allowed");
+                    notifications.addNotification(notification);
```

Comment:
are we covering postaction invocation as part of fsm lateR?

Reply:
@jaimecasero no, I missed the post action and I will add it now